secrets/secrets.go

```diff
@@ -252,10 +262,26 @@ type Vault struct {
 
 // NewSecrets parses and validates the secret JSON provided by the reader.
 func NewSecrets(r io.Reader) (*Secrets, error) {
-	var secretsDocument Document
+	secretsDocument := Document{
+		Secrets: make(map[string]GenericSecret),
+	}
 	err := json.NewDecoder(r).Decode(&secretsDocument)
 	if err != nil {
-		return nil, err
+		switch e := err.(type) {
+		case *fs.PathError:
+			// check if the path is a directory, then assume
+			// the secret provider is Vault CSI
+			if strings.Contains(e.Error(), "is a directory") {
```

Comment:
why are you worrying about directory errors in an `io.Reader`-targeted API?

Reply:
When using vault csi the secret path can be a directory so I have to check for this case and handle the directory differently. Ive moved this into its own helper

Reply:
ive dropped this pattern of error checking and handle dir paths before trying to read